lib/publiccloud/instance.pm

```diff
@@ -364,28 +366,40 @@ sub wait_for_ssh {
             # On boottime test we do hard reboot which may change the instance address
             script_run("ssh-keyscan $args{public_ip} | tee -a ~/.ssh/known_hosts") if (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME'));
 
-            my $output = $self->run_ssh_command(
-                cmd => 'sudo journalctl -b | grep -E "Reached target (Cloud-init|Default|Main User Target)"',
-                proceed_on_failure => 1,
-                username => $args{username});
-            if ($output =~ m/Reached target.*/) {
+            $out = $self->run_ssh_command(cmd => 'sudo systemctl is-system-running --wait', proceed_on_failure => 1, username => $args{username});
+
+            if ($out =~ m/running/) {
+                # result ok print
+                record_info("CHECK SSH", 'wait ssh - system running,' . (time() - $start_time) . 's, System:' . "\n" . $out);
+                $self->journal_upload("/tmp/waitssh_journal.txt", username => $args{username});
+                return $duration;
+            }
+            elsif ($out =~ m/degraded/) {
+                my $failed = $self->run_ssh_command(cmd => 'sudo systemctl --failed', proceed_on_failure => 1, username => $args{username});
+                record_info("CHECK SSH", 'wait ssh - system degraded, ' . (time() - $start_time) . 's, System:' . "\n" . $failed);
+                $self->journal_upload("/tmp/waitssh_journal.txt", username => $args{username});
                 return $duration;
             }
-            elsif ($output =~ m/Permission denied \(publickey\).*/) {
+            elsif ($out =~ m/maintenance|stopping|offline|unknown/) {
+                # check if system status ko
```

Comment:
maybe want to log some record info here with what exactly status was ?

Reply:
The line checking `/Permission denied.../`, inherited by the previous code, indeed it could occurr when some ssh issue could prevent the access to the remote session, therefore I left it there.

Reply:
Instead the line checking the `maintenance|...` is the 3rd status possible, resuting after the `--wait`, below discussed.

Reply:
ok